src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
+// const deer = new Herbivore('Bembi');
+// const panther = new Carnivore('Bagira');
+// const lion = new Carnivore('King');
+// const rabbit = new Herbivore('Max');
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  name = this.name;
+  health = this.health;
+  hidden = false;
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super();
+  }
```

Comment:
The constructor in the `Carnivore` class should pass the `name` and `health` arguments to the `super` call to properly initialize the inherited properties from the `Animal` class.